app/lib/task/cloudcompute/cloudcompute.dart

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) 2021, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// Interface describing an instance.
+abstract class CloudInstance {
+  /// Unique instance name.
+  String get name;
+
+  /// Zone this instance lives in.
+  String get zone;
+
+  /// [DateTime] of when the instance was created.
+  DateTime get created;
+
+  /// State of the instance.
+  InstanceState get state;
+
+  /// Representation for debugging purposes.
+  @override
+  String toString() => 'CloudInstance(${[
+        'name: $name',
+        'zone: $zone',
+        'created: $created',
+        'state: $state',
+      ].join(',')})';
+}
+
+/// Simplified instance state.
+///
+/// In practice we often don't care if an instance _provisioning_ or _staging_.
+/// If an instance is stuck in the [pending] state it's probably because it's
+/// not going to startup.
+///
+/// Similarly, we don't care if an instance is _booting_, _running_, or if the
+/// process running on the instance crashed without terminating the instance.
+/// In all these cases we want delete the instance if it gets stuck long
+/// enought to exhaust allowed execution time.
+///
+/// We also don't care if the instance is _suspending_, _suspended_, _stopping_,
+/// _stopped_ or _terminated_. In these the cases the instance should be
+/// deleted.
+enum InstanceState {
+  /// Instance is pending provisioning or in the process of being provisioned.
+  pending,
```

Comment:
I'm not sure `provisioning` is better, because we are in a state here where we're asking for a VM.

It's entirely possible the request could be denied, then we jump to `terminated`.

I've merged a few states because we don't want to distinguish between too many of them :D